pageserver/src/tenant/timeline.rs

```diff
@@ -3479,6 +3524,10 @@ impl Timeline {
             );
             layers_to_remove.push(Arc::clone(&l));
         }
+        self.wanted_image_layers
+            .lock()
+            .unwrap()
+            .replace(wanted_image_layers);
```

Comment:
A bit uneasy about this lock with all of the other locks, at least `layer_map`, `layer_removal_cs`, perhaps others. Perhaps `layer_removal_cs` should be repurposed to transfer this information altogether, because it's just a lock over unit?

Reply:
Sorry, I do not understand your concern about this extra lock.
It seems to be better to have separate lock rather than try to reuse any of existed locks to avoid false lock conflicts.
This lock is hold for very short time, so it can not somehow cause interleaving with other locks,for  example deadlock.

Reply:
I guess I was thinking that this might introduce deadlocks but of course that's not possible because it's only held for a short while.